### PR TITLE
Implement :edit-command.

### LINKED ISSUE #2453: Edit command using ${editor}
**Feature request:** Provide a way to edit the current command using one's `${editor}`, similar to:

- vim: `c_^F` / `q:`
- bash: `edit-and-execute-command`
- zsh: `edit-command-line`

Qutebrowser already has `edit-url`, which launches `${editor}` to edit a url (default: current url), so this would be similar to that. Call it `edit-command`, for instance.

**Example 1:**

- `: <execute some really long command>`
- Get error (e.g. for a `:bind` or `:set`), or realize I used the wrong search engine or forgot some option or whatever.
- `:<C-p>` (go back to previous command)
- `<C-f>` (open command in editor, where `<C-f>` is bound to `edit-command`)
- fix command, save, quit, and execute command

In this example, `edit-command` is called in command mode, just like `c_^F` in vim.

**Example 2:**

- I have a url in my clipboard, which I want to open in a new tab, but I need to edit the url before opening it.
- `q:` (open editor, where `q:` is bound to `edit-command2`; see below)
- Type `open -t `, paste my url, edit it however I want, save, quit, and execute command

In this example, `edit-command2` is called in normal mode, just like `q:` in vim.

I wrote `edit-command2` to make it clear that it's slightly different from `edit-command`, in the sense that `edit-command` is invoked in command mode, while `edit-command2` is invoked in normal mode (just like `edit-url` currently).

Also, I'd be happy even if there were *only* the command-mode version, because example 2 could just as well go like this:

- `:open -t <paste url>` (but don't execute)
- `<C-f>` (i.e. `<edit-command>`)
- edit the url, save, quit, and execute command

**Rationale:** The point is that editing a command in vim (or whatever) would be way easier/faster than ctrl-arrowing across words or trying to remember emacs-style shortcuts like `<C-a>` and so on.

### PR DESCRIPTION
**Pull request by rcorre**
:edit-command opens the current command line in an editor, and updates
the command line if the editor exits successfully. If --run is passed,
the command is executed when the editor exits sucessfully.

Resolves #2453.

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3306)

### CODE REVIEW

File: qutebrowser/mainwindow/statusbar/command.py [hunk 1]
```diff
@@ -166,6 +166,24 @@ def command_accept(self):
         modeman.leave(self._win_id, usertypes.KeyMode.command, 'cmd accept')
         self.got_cmd[str].emit(prefixes[text[0]] + text[1:])
 
+    @cmdutils.register(instance='status-command', scope='window', maxsplit=0)
+    def edit_command(self, run=False):
+        """Open an editor to modify the current command.
+
+        Args:
+            run: Run the command if the editor exits sucessfully.
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/statusbar/command.py]: sucess -> success

File: qutebrowser/mainwindow/statusbar/command.py [hunk 2]
```diff
@@ -166,6 +166,24 @@ def command_accept(self):
...
+        """
+
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/statusbar/command.py]: Please remove this blank line.

**Comment by rcorre** [comment]
> Probably makes sense to bind ctrl-e in command mode to this by default?

**Comment by The-Compiler** [comment]
Hm, bummer. Well, no default binding then :wink:

File: qutebrowser/mainwindow/statusbar/command.py [hunk 3]
```diff
@@ -166,6 +166,23 @@ def command_accept(self):
         modeman.leave(self._win_id, usertypes.KeyMode.command, 'cmd accept')
         self.got_cmd[str].emit(prefixes[text[0]] + text[1:])
 
+    @cmdutils.register(instance='status-command', scope='window', maxsplit=0)
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/statusbar/command.py]: Why `maxsplit=0`? This prevents splitting of positional arguments (like with `:bind`) - while we don't have any here, I guess that was unintentional?

Comment (rcorre) [reply qutebrowser/mainwindow/statusbar/command.py]: oops, copy-paste error. Fixed.

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**